src/session.rs

```diff
@@ -277,6 +277,15 @@ impl Session {
         )
     }
 
+    pub fn shared_command<'a, S: Into<Cow<'a, str>>>(self: std::sync::Arc<Self>, program: S) -> OwnedCommand<std::sync::Arc<Self>> {
+        self.shared_raw_command(&*shell_escape::unix::escape(program.into()))
+    }
+
+    pub fn shared_raw_command<S: AsRef<OsStr>>(self: std::sync::Arc<Self>, program: S) -> OwnedCommand<std::sync::Arc<Self>> {
```

Comment:
I would like to also have another generic version of this that takes `this: impl AsRef<Session> + Clone`

Reply:
I'm not sure `impl` can be used this way, and I'm reasonably certain it it's not supported (yet) for a `self`. It could be a free or factory function, but in that case maybe it should be a public constructor for `OwnedCommand` instead?

Reply:
[Currently valid list of self types](https://doc.rust-lang.org/reference/items/associated-items.html#methods) is `Self`, reference, box, rc, arc, and pin. So yeah no impl as an instance method.

Reply:
You can make it an associated function instead `this: impl AsRef<Session> + Clone` is valid.

Reply:
Oh I see that's what you meant, sorry I misunderstood.

Should it just replace the Arc version then? As it kinda obviates it, and if we call it "to_command" or something like that, even with an explicit `Arc` wrapper it's a lot shorter still.

Reply:
You could keep it as a convenient method, or remove it if you think it's unnecessary.

Reply:
I'll be honest here, I don't use the crate, I just saw the issue during jonhoo's stream and figured I had an idea which could solve the problem with not too much code duplication (or macro).

So while I do have my own sensibilities about code and APIs, I'd *much* rather go with what you, jon, and other actual users of the crate (maybe the other maintainers of openssh-sftp-client since that's where things started?) would prefer.

Reply:
In that case, let's keep the arc_command, AFAIK many do use it from multi-threaded async context with Arc.

Reply:
OK if it's a common use case it makes sense to have a convenient shortcut. And the associated function does look pretty good in the test.

I made the arc versions just be convenience shortcuts over the associated function as that seems simpler / clearer.